src/SignalR/clients/csharp/Client.SourceGenerator/src/HubProxyGenerator.Parser.cs

```diff
@@ -0,0 +1,198 @@
+
+                    var members = hubSymbol.GetMembers()
+                        .Where(member => member.Kind == SymbolKind.Method)
+                        .Select(member => (IMethodSymbol) member)
+                        .Concat(hubSymbol.AllInterfaces.SelectMany(x => x
+                            .GetMembers()
+                            .Where(member => member.Kind == SymbolKind.Method)
+                            .Select(member => (IMethodSymbol) member)));
+
+                    // Generate spec for each method
+                    foreach (var member in members)
+                    {
+                        var methodSpec = new MethodSpec
+                        {
+                            Name = member.Name, FullyQualifiedReturnTypeName = member.ReturnType.ToString()
+                        };
+
+                        if (member.ReturnType is INamedTypeSymbol {Arity: 1} rtype)
+                        {
+                            methodSpec.InnerReturnTypeName = rtype.TypeArguments[0].ToString();
+                        }
+
+                        if (member.ReturnType is INamedTypeSymbol {Arity: 1, Name: "Task"} a
+                            && a.TypeArguments[0] is INamedTypeSymbol {Arity: 1, Name: "ChannelReader"} b)
+                        {
+                            methodSpec.Stream = StreamSpec.ServerToClient & ~StreamSpec.AsyncEnumerable;
+                            methodSpec.InnerReturnTypeName = b.TypeArguments[0].ToString();
+                        }
+                        else if (member.ReturnType is INamedTypeSymbol {Arity: 1, Name: "IAsyncEnumerable"} c)
+                        {
+                            methodSpec.Stream = StreamSpec.ServerToClient | StreamSpec.AsyncEnumerable;
+                            methodSpec.InnerReturnTypeName = c.TypeArguments[0].ToString();
+                        }
+                        else
+                        {
+                            methodSpec.Stream = StreamSpec.None;
+                        }
+
+                        // Generate spec for each argument
+                        foreach (var parameter in member.Parameters)
+                        {
+                            var argumentSpec = new ArgumentSpec
+                            {
+                                Name = parameter.Name, FullyQualifiedTypeName = parameter.Type.ToString()
+                            };
+
+                            methodSpec.Arguments.Add(argumentSpec);
+
+                            switch (parameter.Type)
+                            {
+                                case INamedTypeSymbol {Arity: 1, Name: "ChannelReader"} when
+                                    methodSpec.Stream.HasFlag(StreamSpec.ServerToClient) &&
+                                    methodSpec.Stream.HasFlag(StreamSpec.AsyncEnumerable):
+                                    _context.ReportDiagnostic(Diagnostic.Create(
```

Comment:
Only because I might have had a narrower interpretation of how streaming calls work :)

I was of the understanding that a streaming call is only using `IAsyncEnumerable` or `ChannelReader` as "frontend" but not a mix. I'll remove these cases / make it less restrictive.